test/e2e/nsx_ipblocksinfo_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+var (
+	ipBlocksInfoCRDName = "ip-blocks-info"
+	defaultOrg          = "default"
+	defaultProject      = "project-quality"
+	defaultVPCProfile   = "default"
+)
+
+func TestIPBlocksInfo(t *testing.T) {
+	t.Run("case=InitialIPBlocksInfo", InitialIPBlocksInfo)
+	t.Run("case=CustomIPBlocksInfo", CustomIPBlocksInfo)
+}
```

Comment:
do we need a case when one vpc(vpcconfiguration) deleted, the ipblockinfo updated?

Reply:
Yes, the defer func in CustomIPBlocksInfo case checks this
```
        defer func() {
		// Delete VPCNetworkConfigurations and check
		err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Delete(context.TODO(), vpcConfigName, metav1.DeleteOptions{})
		require.NoError(t, err)
		assertIPBlocksInfo(t, privateTGWIPCIDRs, externalIPCIDRs)
	}()
```